### PR TITLE
Bootstrap switchboard

### PR DESCRIPTION
**Pull request by max-cura**

### CODE REVIEW

File: connector/ws/src/bin/wscx.rs
Comment (lschuermann) [inline connector/ws/src/bin/wscx.rs:53]: High-level remark: eventually we want to just use TLS and authenticate through an `Authorization: Bearer <token>` header, as that's significantly less complex than this machinery here. Given it's already built, we can roll with this for now.

File: switchboard/README.md
Comment (lschuermann) [inline switchboard/README.md:6]: This seems no longer current.

File: switchboard/scripts/README.md
Comment (lschuermann) [inline switchboard/scripts/README.md:3]: I don't know whether we need this README at all. If all the scripts have a help message prompt or a comment at the top of the respective file, I think that's enough docs for now.

File: switchboard/scripts/setup.sh
Comment (lschuermann) [inline switchboard/scripts/setup.sh:1]: Remove?

File: switchboard/scripts/sqlx-precommit-setup.sh
Comment (lschuermann) [inline switchboard/scripts/sqlx-precommit-setup.sh:1]:
```suggestion
#!/usr/bin/env sh
```

If only to avoid diverging from how the other scripts work.

File: switchboard/sql/DEBUG_VALUES.sql
Comment (lschuermann) [inline switchboard/sql/DEBUG_VALUES.sql:1]: I think `FIXTURES` is a more standard-term for what this file represents.

File: switchboard/switchboard/Sec-Todo.txt
Comment (lschuermann) [inline switchboard/switchboard/Sec-Todo.txt:1]: Might be better to keep this in a tracking issue on GitHub?

File: switchboard/switchboard/config.example.toml
Comment (lschuermann) [inline switchboard/switchboard/config.example.toml:1]: I think this config looks good so far. #6 will revisit this file and standardize configuration paradigms across the repository.

File: switchboard/switchboard/src/bin/swx.rs
Comment (lschuermann) [inline switchboard/switchboard/src/bin/swx.rs:23]:
```suggestion
    Serve(tml_switchboard::server::serve::ServeConfig),
```

And then define the `ServeConfig` struct analog to the above:
```rust
struct ServeConfig {
    /// Path to server configuration file
    #[arg(short = 'c', long = "cfg", env = "TML_CFG")]
    cfg: PathBuf,
},
```

This would increase cohesion within the implementation of these (independent) subcommands and reduce coupling between the `swx` tool and the respective subcommands.

File: switchboard/switchboard/src/schemas/switchboard_supervisor.rs
Comment (lschuermann) [inline switchboard/switchboard/src/schemas/switchboard_supervisor.rs:1]: This less a schema than it is an API (type) specification, right?

Instead of having other crates depend on the `tml_switchboard` crate, let's try to have all these common type in the `treadmill-rs` crate instead. This avoids pulling in spurious dependencies and code that's unrelated to other users of these types.

In general, if something's representing a contract between two components, is generally useful, and doesn't have any dependencies (specific to a particular component), it's probably a good idea to extract it to the `treadmill-rs` base crate.

File: switchboard/switchboard/src/server/session.rs
Comment (lschuermann) [inline switchboard/switchboard/src/server/session.rs:478]:
```suggestion
// -- Functions for working with sessions
```

**Review by lschuermann — CHANGES_REQUESTED** [review]
This is looking really good and like a solid foundation for more directed follow-up PRs.

In particular the authentication and privilege submodules contain a bunch of engineering and scaffolding already. While that's a great start, we'll need to see how well this ends up working in practice. We shouldn't hesitate to iterate on these components later on.

I skimmed some of the more complex authentication & privilege code. It implements the general design that we talked about before, but I'll have to do a more in-depth review in the future. To bootstrap the switchboard development I would not be opposed to merging this initial version, and then going from there.

I did not do an in-depth review of the database schema yet. I'd propose breaking this out into its own separate issue (#5) and follow-up PR respectively.

File: switchboard/switchboard/src/schemas/switchboard_supervisor.rs
Comment (max-cura) [reply switchboard/switchboard/src/schemas/switchboard_supervisor.rs:1]: Waiting on #12; once that's in, I'll merge switchboard's API stuff into treadmill-rs.

File: switchboard/switchboard/Sec-Todo.txt
Comment (max-cura) [reply switchboard/switchboard/Sec-Todo.txt]: Created #13.

File: switchboard/README.md
Comment (max-cura) [reply switchboard/README.md]: Fixed in 86acf4d.

**Review by lschuermann — APPROVED** [review]
This looks great! Thanks for building a solid foundation for the Switchboard component!

**Comment by lschuermann** [comment]
(FWIW, I fully expect the queued merge to fail because we just merged #12, just want to make sure our pipeline actually works as intended)

**Comment by max-cura** [comment]
I think (?) that's expected... the changes in #12 only modify `rest` and `sse`, and I haven't touched those in supervisor yet.
So i think the changes should be self-contained...

**Pull request merged**
